src/libsync/owncloudpropagator.h

```diff
@@ -261,6 +261,9 @@ class OWNCLOUDSYNC_EXPORT PropagateDirectory : public PropagatorJob
     virtual JobParallelism parallelism() Q_DECL_OVERRIDE;
     virtual void abort() Q_DECL_OVERRIDE
     {
+        // Abort first job and sub jobs
+        // Finished abort will be announced via abortFinished()
+        // (look contructor implementation)
```

Comment:
(Maybe you can do the connection here.)

Here as well you need to have a counter to make sure that both jobs sent their abortFinished.